src/sidebar/store/modules/session.js

```diff
@@ -1,20 +1,26 @@
 import * as util from '../util';
 
+/**
+ * A dummy profile returned by the `profile` selector before the real profile
+ * is fetched.
+ */
+const blankProfile = {
```

Comment:
I imagine it's related to the "cascading" situation you referenced in the PR's description, but can you help me grok why this can't be the initial value of `profile` on store initialization? Is it that the selectors have a hard time differentiating between this object and a "real" profile as returned by the service (I do get that you need to be able to determine if a real profile has been successfully fetched)? I'm wondering if it's possible to use this as the initial value for `profile` but perhaps have an additional property in the state object that flags whether a profile has ever been retrieved? Just musing.

Reply:
Yes, something like that would work. A variation on the above would be to make `hasFetchedProfile` just test whether `store.state.profile === blankProfile`.